src/device/msccl_kernel_impl.h

```diff
@@ -372,7 +372,7 @@ __device__ __forceinline__ void mscclRunInterpreter(
           return;
       }
       if (t->hasDependence && tid == nthreads-1)
-        __atomic_store_n(&mscclFlags[bid].flag, (uint64_t) COMPUTE_FLAG(workIndex, iter, step), ((t->type == MSCCL_REDUCE || t->type == MSCCL_RECV) && (t->dstBuffer != MSCCL_SCRATCH_BUFFER)) ? __ATOMIC_RELEASE : __ATOMIC_RELAXED);
```

Comment:
Should a comment be added here explaining this change?

Reply:
Sure. Out of place all reduce adds sending from output buffer, thus need cache flush/invalidate when this happens

Reply:
The PR looks good. There are some test failures that need to be checked before merging.

Reply:
I guess you mean gfx906 failures? These have been there for months. I think re-run will pass